sycl/doc/extensions/experimental/sycl_ext_oneapi_graph.asciidoc

```diff
@@ -1702,12 +2089,335 @@ Exceptions:
+<<external-event-dependencies, one-time>> or <<dynamic-events, updatable between
+executions>>.
+
+==== Node-Level Execution Events [[node-execution-events]]
+
+Events representing the completion of an individual node inside a given
+executable graph can be obtained using
+`command_graph<graph_state::executable>::get_event(node)`. These events can then
+be waited on or used as dependencies for eager SYCL operations outside of
+graphs. These events may be useful for operations which may be infrequent and
+depend only on some intermediate results of work being done in the graph.
+
+[source, c++]
+....
+sycl::event ExecutionEvent = ExecGraph.get_event(SomeNode);
+
+Queue.submit((sycl::handler& CGH)
+  {
+    CGH.depends_on(ExecutionEvent);
+    CGH.parallel_for(...);
+  });
+
+// The above operation will only execute once SomeNode has finished executing
+// inside ExecGraph
+Queue.ext_oneapi_graph(ExecGraph);
+....
+
+These events represent only the current execution of a given executable graph on
+a device. These events are not unique per execution of a graph. If a graph is
+enqueued multiple times before using one of these events (for example as a
+dependency to an eager SYCL operation or a host wait), it is undefined which
+specific execution of a graph the event will represent. If a dependency on a
+specific graph execution is required this ordering must be enforced by the
+application to ensure there is only a single graph execution in flight when
+using these events.
+
+For example:
+
+[source, c++]
+----
+
+sycl::event NodeExecutionEvent;
+sycl::event GraphCompletionEvent;
+
+for (size_t i = 0; i < GraphIterations; i++){
+  // Obtain the node execution event for the graph
+  NodeExecutionEvent = ExecGraph.get_event(SomeNode);
+
+  // Enqueue the graph for execution
+  sycl::event GraphCompletionEvent = Queue.ext_oneapi_graph(ExecGraph);
+  // Use the event, to wait on the host and perform some intermediate
+  // host-work once that node has completed
+  ExecutionEvent.wait();
+  DoSomethingOnHost();
+  // Wait on the graph finishing to ensure it is complete before the next use of
+  // a node-level execution event
+  GraphCompletionEvent.wait_and_throw();
+}
+----
+
+These events cannot be used to define dependencies between graphs. These should
+instead be defined as described in <<inter-graph-dependencies, this section>>.
+
+==== Adding External Event Dependencies To Graphs [[external-event-dependencies]]
+
+Events from eager SYCL submissions can be passed as dependencies to graph nodes
+to create dependencies at graph execution time between these eager
+operations and specific graph nodes. This is done in the same way as creating
+dependencies between graph nodes using events, for example:
+
+[source, c++]
+----
+// Submit an eager SYCL operation
+event ExternalEvent = SyclQueue.submit(...);
+
+// Record a graph node which depends on this external event
+Graph.begin_recording(SyclQueue);
```

Comment:
```suggestion
Graph.begin_recording(SyclQueue or AnotherSyclQueue);
```

it could be in different sycl queues

Reply:
Does this need to be specified? There are no restrictions on recording from different queues, or restrictions across queues with normal SYCL submissions. I feel like the user can infer from the lack of this restriction in any of the function definitions that this is fine, and repeating it in every single example or description is unnecessary.